apps/settings/js/panels/wifi_select_certificate_file/panel.js

```diff
@@ -71,38 +71,87 @@ define(function(require) {
         li.appendChild(anchor);
 
         anchor.onclick = () => {
-          DialogService.show('wifi-enterCertificateNickname', {
-            certificateName: certificateName
-          }).then((result) => {
-            var type = result.type;
-            var value = result.value;
-
-            if (type === 'submit') {
-              var certRequest =
-                wifiManager.importCert(file, '', value.nickname);
-
-              // Gray out all item of certificate files
-              // since we are importing other file.
-              this._setCertificateItemsEnabled(false);
-              certRequest.onsuccess = () => {
-                // direct dialog to "wifi-manageCertificates"
-                SettingsService.navigate('wifi-manageCertificates');
-              };
-
-              certRequest.onerror = () => {
-                DialogService.alert('certificate-import-failed-description', {
-                  title: 'certificate-import-failed'
-                }).then(() => {
-                  // Re-enable all items of certificate files
-                  // since import file process is completed.
-                  this._setCertificateItemsEnabled(true);
-                });
-              };
-            }
-          });
+          this._importCertificate(file);
         };
         return li;
       },
+      _importCertificate: function(file) {
+        var certificateName = this._parseFilename(file.name);
+        var isPasswordRequired =
+          this._isPasswordRequired(this._parseExtension(file.name));
+        var password, nickname;
+
+        (() => {
+          this._setCertificateItemsEnabled(false);
+          if (isPasswordRequired) {
+            return this._requestPassword();
+          } else {
+            return Promise.resolve(null);
+          }
+        })().then((result) => {
+          if (isPasswordRequired && !result) {
+            return Promise.reject('invalid-password');
+          } else {
+            password = result;
+            return this._requestNickName(certificateName);
+          }
+        }).then((result) => {
+          nickname = result;
+          return new Promise((resolve, reject) => {
+            var req = wifiManager.importCert(file, password, nickname);
+            req.onsuccess = resolve;
+            req.reject = onerror;
```

Comment:
I use following form to bypass error message from domrequest.onerror() to promise.reject() to test Bug 1163031.
```
req.onerror = function (e) {
  reject(e.target.error.name);
}
```